src/tests/swap.c

```diff
@@ -278,7 +275,7 @@ void test_swap_speed2 (const char *buf)
   TIME_IT (intrin_byteswap,      (buf, swap_size/4), loops);
 }
 
-void __watcall sigill_handler2 (int sig)
+void sigill_handler2 (int sig)
```

Comment:
Strange. According to `signal.h`, a Watcom signal-handler is:
`typedef void (_WCCALLBACK *__sig_func)( int );` and  `_WCCALLBACK` is `__watcall`.

But dropping `__watcall`, it works.

Reply:
_WCCALLBACK is not only about calling convention.
__watcall is default that not need to be specified, it has sense if you change default calling convention what is not current and not your case

Reply:
So if I do:
```diff
diff --git a/src/tests/swap.c b/src/tests/swap.c
index 07fc157..fad93d5 100644
--- a/src/tests/swap.c
+++ b/src/tests/swap.c
@@ -22,7 +22,7 @@
   #define TEST_UNDEF_OPCODE

 #elif defined(__WATCOMC__) && defined(__386__)
-//#define INVD_CACHE
+  #define INVD_CACHE
   #define TEST_UNDEF_OPCODE
 #endif
```
Compile and run `swap.exe`, I get an exception. Either the `sigill_handler()` in `sysdep.h` is wrong or this:
```c
  #pragma aux _invd_cache = \
          "db 0Fh, 08h";
```

is wrong. But Im not familar with Watcom ASM.